src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js

```diff
@@ -76,7 +88,56 @@ export default class UserSettingsContainer extends Component {
       router.push(nextPage);
     }
     untouch('preferredName');
-  };
+  }
+
+  uploadPicture(pictureFile) {
```

Comment:
i think a faster, simpler solution might be setting up an endpoint for s3 to call post-upload.

Right now, the client -> S3 -> client -> Parabol -> client.
I propose: client -> S3 -> Parabol -> client.
Alternatively, and this depends on your thoughts about deterministic file names, we go client -> Parabol -> s3 -> Parabol -> client. The reasoning being that we don't want the client to be able to name the file otherwise they could purposefully cause a collision.